File: library/src/main/java/eu/sesma/library/LibraryMain.java
package eu.sesma.library;

import android.util.Log;

import javax.inject.Inject;

public class LibraryMain {

    private final static String TAG = LibraryMain.class.getSimpleName();

    @Inject
    LibraryCollaborator collaborator;


    public void initialize() {

        DaggerLibraryComponent.create().inject(this);

        Log.d(TAG, "Hello collaborator " + collaborator.greet());
    }

}

File: module_detail/src/main/java/eu/sesma/main/LibraryMain.kt
package eu.sesma.main

import android.util.Log

import javax.inject.Inject

class LibraryMain {

    companion object {
        private val TAG = LibraryMain::class.java.getSimpleName()
    }

    @Inject
    lateinit var collaborator: LibraryCollaborator

    fun initialize() {

        DaggerLibraryComponent.create().inject(this)

        Log.d(TAG, "Hello collaborator " + collaborator.greet())
    }
}
